src/main/java/hudson/plugins/analysis/core/HealthAwarePublisher.java

```diff
@@ -145,12 +158,12 @@ public HealthAwarePublisher(final String healthy, final String unHealthy,
      *             if the user canceled the build
      */
     @Override
-    protected boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final PluginLogger logger)
+    protected boolean perform(Run<?, ?> run, FilePath workspace, Launcher launcher, PluginLogger logger)
```

Comment:
Not clear that you are maintaining strict binary compatibility with these changes... though given the complexity of this class it could be rather tricky